src/components/Countdown/index.tsx

```diff
@@ -0,0 +1,55 @@
+import { FC, useEffect, useState } from "react"
+import {
+  dateToUnixTimestamp,
+  ONE_SEC_IN_MILISECONDS,
+  dateAs,
+  addLeadingZero,
+} from "../../utils"
+import { H4 } from "../Typography"
+
+export interface CountdownProps {
+  targetDateInUnix: number
+  onComplete?: (targetDateInUnix: number) => void
+  withLeadingZeroes?: boolean
+  children: (
+    days: string,
+    hours: string,
+    minutes: string,
+    seconds: string
+  ) => JSX.Element
+}
+
+export const Countdown: FC<CountdownProps> = ({
+  targetDateInUnix,
+  onComplete,
+  withLeadingZeroes = true,
+  children,
+}) => {
+  const [diff, setDiff] = useState(targetDateInUnix - dateToUnixTimestamp())
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      const diff = targetDateInUnix - dateToUnixTimestamp()
+      if (diff === 0) {
+        if (onComplete) {
+          onComplete(targetDateInUnix)
+        }
+        clearInterval(interval)
+      }
+
+      setDiff(diff)
+    }, ONE_SEC_IN_MILISECONDS)
+
+    return () => clearInterval(interval)
+  }, [targetDateInUnix])
+  let { days, hours, minutes, seconds } = dateAs(diff)
+
+  if (withLeadingZeroes) {
+    days = addLeadingZero(Number(days))
+    hours = addLeadingZero(Number(hours))
+    minutes = addLeadingZero(Number(minutes))
+    seconds = addLeadingZero(Number(seconds))
+  }
+
+  return children(days, hours, minutes, seconds)
```

Comment:
Maybe we could extract it to a separate hook and export the newly created hook from our components lib so anyone can choose which option prefer. Eg I can imagine a case when based on the returned values from hook someone want to do something more so in that case hook fits better instead of a Component.